src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java

```diff
@@ -327,13 +327,26 @@ boolean onError(TDSReader tdsReader) throws SQLServerException
                 return false;
             }
 
-            boolean onDone(TDSReader tdsReader) throws SQLServerException
-            {
-                // When initializing client-cursored ResultSets, a DONE token
-                // following the column metadata indicates an empty result set.
-                rowCount = 0;
-                return false;
-            }
+			boolean onDone(TDSReader tdsReader) throws SQLServerException {
+				// When initializing client-cursored ResultSets, a DONE token
+				// following the column metadata indicates an empty result set.
+				rowCount = 0;
+
+				// Continue to read the error message if DONE packet has error flag
+				int packetType = tdsReader.peekTokenType();
+				if (TDS.TDS_DONE == packetType) {
+					short status = tdsReader.peekStatusFlag();
```

Comment:
Function of methods in TDSReader are restricted to read operation, it doesn't do any data validation, it's good to keep it that way.